assign_rights/views.py

```diff
@@ -28,24 +28,74 @@ def get_context_data(self, **kwargs):
         return context
 
 
-def RightsShellListView(ListView):
+class RightsShellListView(LoginRequiredMixin, ListView):
     '''browse and search rights shells'''
-    pass
+    queryset = RightsShell.objects.all()
+    template_name = "rights/list.html"
 
 
-class RightsShellCreateView(CreateView):
-    '''create rights shells'''
-    pass
+class RightsShellCreateView(LoginRequiredMixin, CreateView):
+    model = RightsShell
+    template_name = "rights/manage.html"
+    form_class = RightsShellForm
+    success_url = None
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        if self.request.POST:
+            context['rights_granted'] = RightsGrantedFormSet(self.request.POST)
+        else:
+            context['rights_granted'] = RightsGrantedFormSet()
+        return context
+
+    def form_valid(self, form):
+        context = self.get_context_data(form=form)
+        rights_granted = context['rights_granted']
+        if rights_granted.is_valid():
+            response = super().form_valid(form)
+            rights_granted.instance = self.object
+            rights_granted.save()
+            return response
+        else:
+            return super().form_invalid(form)
+
+    def get_success_url(self):
+        return reverse_lazy("rights-detail", kwargs={"pk": self.object.pk})
 
 
-class RightsShellDetailView(DetailView):
+class RightsShellDetailView(LoginRequiredMixin, DetailView):
     '''view a rights shell'''
-    pass
+    queryset = RightsShell.objects.all()
```

Comment:
again, you can just do
```suggestion
    model = RightsShell
```

which in this case feels a little more intuitively right?